File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/SearchNotesViewModel.java
package com.intkhabahmed.smartnotes.viewmodels;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.intkhabahmed.smartnotes.database.NoteRepository;
import com.intkhabahmed.smartnotes.models.Note;

import java.util.List;

public class SearchNotesViewModel extends ViewModel {
    private LiveData<List<Note>> notes;

    SearchNotesViewModel(String title, int trashed) {
        if (notes == null) {
            setNotes(title, trashed);
        }
    }

    public void setNotes(String title, int trashed) {
        notes = NoteRepository.getInstance().getNotesByTitleAndAvailability(title, trashed);
    }

    public LiveData<List<Note>> getNotes() {
        return notes;
    }
}

File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/SearchNotesViewModel.kt
package com.intkhabahmed.smartnotes.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList

import com.intkhabahmed.smartnotes.database.NoteRepository
import com.intkhabahmed.smartnotes.models.Note

class SearchNotesViewModel internal constructor(title: String?, trashed: Int) : ViewModel() {
    lateinit var notes: LiveData<PagedList<Note>>

    init {
        setNotes(title, trashed)
    }

    fun setNotes(title: String?, trashed: Int) {
        notes = LivePagedListBuilder(NoteRepository.instance!!.getNotesByTitleAndAvailability(title, trashed),
                PagedList.Config.Builder().setPageSize(10).setInitialLoadSizeHint(20).setEnablePlaceholders(true).build()).build()
    }
}
